### ISSUE TITLE
Namespace collision with PusherSwift CocoaPods pod due to Pusher classes declared in Beacon header

### ISSUE THREAD

**Issue by jgenden** [helpscout/beacon-ios-sdk#167]
**Technical Details:**

- Beacon install method: 
    [ ] Carthage, 
    [X] CocoaPods, 
    [ ] Swift Package Manager,
    [ ] Manual
- SDK version: Beacon 2.2.3 Cocoapods
-
- iOS OS version: N/A
- Device make/model: N/A
- Impacted user count: N/A

**Actual outcome**

Project failed to compile when including both PusherSwift (10.0.0) and Beacon (2.2.3) via cocoapods. Headers for Pusher are declared in the Beacon header, which causes errors when using the latest version of PusherSwift.  Example error messages:

```'PusherClientOptions' has different definitions in different modules
'GlobalChannel' has different definitions in different modules
'PusherChannel' has different definitions in different modules
```

And several more.

**Expected outcome**

Beacon should be able to compile along with the latest version of PusherSwift. If possible, PusherSwift should be a dependency of Beacon rather than built into the binary. 

**Steps to reproduce**

include both `PusherSwift` and `Beacon` in your Podfile and import both.

**Comment by codenameDuchess** [comment]
Hi @jgenden,

Thanks for creating an issue and sorry for the trouble. Would you please provide us with a bit more information, or perhaps a copy of your podfile.lock so that we can investigate further?  I've set up a local project pulling in Beacon with CocoaPods and the latest version of Pusher and I've not been able recreate this issue. 

I was able to import both Beacon and PusherSwift in my view controller without any compilation errors. I am using Xcode 12.5.1 and CocoaPods 1.10.2 on MacOS BigSur, which versions are you using?

These are the combinations I've tried dependency wise, both compiled and ran without issue:
<img width="420" alt="Screen Shot 2021-08-23 at 1 05 14 PM" src="https://user-images.githubusercontent.com/4019969/130487998-f09ed9f7-17bb-45ed-9da6-c4aa05fcfbad.png">
<img width="420" alt="Screen Shot 2021-08-23 at 12 58 09 PM" src="https://user-images.githubusercontent.com/4019969/130488074-42f2c822-62bd-4db8-9058-c24a89ce41fa.png">

Perhaps with a bit more information I can help you track this issue down. Thanks again!

**Comment by jgenden** [comment]
@codenameDuchess thank you so much for looking into this. I'll take a look at this a little later today.

**Comment by jgenden** [comment]
@codenameDuchess Thanks again for taking a look. I'm noticing that I don't hit any errors until I actually attempt to invoke some `Pusher` code - did you call anything in Pusher? You can do something like this:

**Comment by codenameDuchess** [comment]
@jgenden That did make the difference for me and I'm able to see the same compilation error now. Thanks for reporting and helping us reproduce this issue. I've created an [internal tracking ticket](https://helpscout.atlassian.net/browse/IOSSDK-495) for us to address this. 

As a temporary work around you could, if possible, switch to the older version of PusherSwift we currently include in our Beacon binary via `pod 'PusherSwift', '~> 8.0'` in your podfile. Once again, sorry for the trouble, and we'll let you know once we have a solution.

**Comment by jgenden** [comment]
appreciated! I'll try it out with the old version to get up and running for the short term. Thanks again.

**Comment by ScottPetit** [comment]
Hey @jgenden, I just wanted to reach and let you know that we've released a new version (2.2.4) that no longer has collisions with newer versions of PusherSwift.  You should be good to update Beacon and PusherSwift if you'd like.  Thanks for filing the issue and If you want to see what else in this release feel free to check out our [Changelog](https://github.com/helpscout/beacon-ios-sdk/blob/master/CHANGELOG.md).

**Comment by jgenden** [comment]
@ScottPetit much appreciated! I'll give it a shot shortly. Thanks very much for being so responsive on here.
